enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,483 @@
+
+![Alt text](assets/community-operators.png?raw=true "community-operators")
+
+The way we describe the intent of a bundle to be on a particular channel in a package is by hardcoding that information in the [bundle annotations](https://github.com/operator-framework/operator-registry/blob/release-4.6/docs/design/operator-bundle.md#bundle-annotations) inside the bundle metadata itself. However, the channel that the bundle wants to be in is more of a package level information. Hardcoding channel information inside a bundle restricts a bundle in a particular channel once the bundle is built and loaded into an index. This makes it difficult to change channel info/any other package level info inside the index retroactively. The only way to change any package level information currently is to rebuild the bundles and then republish the index. This is done by making additive changes to the index using the`opm` tool. 
+
+Furthermore, a lot of the rules about the structure of an index are implicitly documented in various sources that are scattered. A best effort is being made to keep the rules updated in the form of a library in the [api repo](https://github.com/operator-framework/api/tree/master/pkg/validation) that is imported by tools like `opm`, `operator-courier` etc that perform index validation. However, like all tools that aspire to provide data validation, these tools, or api libraries that these tools import rules from, have to be constantly maintained to keep up with the latest rule changes, while also maintaining older rules in previous releases. This introduces a lot of overhead in ensuring that olm is easy to use by users, while also leaving windows for lapses due to mis-coordination between different tools across different versions.
+
+Since indexes are manipulated by making additive changes to it, the way an index can be reproduced today is my managing a knowledge base of the series of changes made to the index (i.e storing the `opm` commands chronologically) and by mananging a set of intermediary artifacts (images of the index with their corresponding digest etc). This introduces a lot of overhead in maintaining indexes, and thereby has significantly burdened the `opm` tool with feature requests to add an increasing number of subcommands to allow index authors to manipulate their indexes in various ways. 
+
+### Goals
+
+A human consumable representation of pacakages in an index and the entire index itself in the form of json/yaml will: 
+
+1. Allow for editing of package level information without rebuilding and republishing artifacts(remove bunlde/s from a channel, switch bundle/s to a different channel etc).
```

Comment:
i think this also needs to allow additional upgrade edges to be defined and removed.  "i previously published v2.10 and said it could upgrade from v1.10.  I now want v2.10 to be able to also upgrade from v1.11, and i do not want to rebuild/republish a v2.10 or v2.11 to make that happen"